### PR TITLE
[FLINK-18194][walkthrough] Update Table API Walkthrough 

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change

Removes the table walkthrough archetype (which is being replaced with a skeleton project in flink playgrounds) and updates the docs accordingly. 

## Brief changelog

- Remove table walkthrough archetypes
- Remove corresponding e2e tests
- Update documentation

Please reference https://github.com/apache/flink-playgrounds/pull/13 to see the corresponding code.

### CODE REVIEW

File: docs/try-flink/table_api.md [hunk 1]
```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
...
-In this tutorial, you will learn how to build a continuous ETL pipeline for tracking financial transactions by account over time.
-You will start by building your report as a nightly batch job, and then migrate to a streaming pipeline.
+In this tutorial, you will learn how to build a real-time dashboard to track financial transactions by account.
+The pipeline will read data from Kafka and write the results to MySQL visualized via Grafana.
 
 ## Prerequisites
 
-This walkthrough assumes that you have some familiarity with Java or Scala, but you should be able to follow along even if you are coming from a different programming language.
-It also assumes that you are familiar with basic relational concepts such as `SELECT` and `GROUP BY` clauses. 
+This walk-through assumes that you have some familiarity with Java or Scala, but you should be able to follow along even if you come from a different programming language.
```
Comment (twalthr) [inline docs/try-flink/table_api.md:39]: nit: wikipedia says `walkthrough` is more common

File: docs/try-flink/table_api.md [hunk 2]
```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
...
 
 #### The Execution Environment
 
-The first two lines set up your `ExecutionEnvironment`.
-The execution environment is how you can set properties for your Job, specify whether you are writing a batch or a streaming application, and create your sources.
-This walkthrough begins with the batch environment since you are building a periodic batch report.
-It is then wrapped in a `BatchTableEnvironment` to have full access to the Table API.
+The first two lines set up your `TableEnvironment`.
+The table environment is how you can set properties for your Job, specify whether you are writing a batch or a streaming application, and create your sources.
+This walkthrough creates a standard table environment that uses the streaming runtime.
```
Comment (twalthr) [inline docs/try-flink/table_api.md:112]: nit: we are always using a streaming runtime but maybe we are configuring the environment in a "streaming mode"?

File: docs/try-flink/table_api.md [hunk 3]
```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
...
-{% endhighlight %}
-</div>
-</div>
-
 
 #### Registering Tables
 
-Next, tables are registered in the execution environment that you can use to connect to external systems for reading and writing both batch and streaming data.
-A table source provides access to data stored in external systems; such as a database, a key-value store, a message queue, or a file system.
+Next, tables are registered in the environment that you can use to connect to external systems for reading and writing both batch and streaming data.
```
Comment (twalthr) [inline docs/try-flink/table_api.md:121]: `in the current catalog` for completeness and for advertising catalog support

File: docs/try-flink/table_api.md [hunk 4]
```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
...
-tEnv.registerTableSource("transactions", new BoundedTransactionTableSource());
-tEnv.registerTableSink("spend_report", new SpendReportTableSink());
-{% endhighlight %}
-</div>
-
-<div data-lang="scala" markdown="1">
-{% highlight scala %}
-tEnv.registerTableSource("transactions", new BoundedTransactionTableSource)
-tEnv.registerTableSink("spend_report", new SpendReportTableSink)
+{% highlight sql %}
```
Comment (twalthr) [inline docs/try-flink/table_api.md:126]: shouldn't this be `highlight java`?

File: docs/try-flink/table_api.md [hunk 5]
```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
...
-</div>
 
-<div data-lang="scala" markdown="1">
-{% highlight scala %}
-env.execute("Spend Report")
-{% endhighlight %}
-</div>
-</div>
+One of Flink's unique properties is that it provides consistent semantics across batch and streaming.
+This means you can develop and test applications in batch mode on static datasets, and deploy to production as streaming applications!
```
Comment (twalthr) [inline docs/try-flink/table_api.md:186]: remove `!`?

File: docs/try-flink/table_api.md [hunk 6]
```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
...
 
 ## Attempt One
 
 Now with the skeleton of a Job set-up, you are ready to add some business logic.
 The goal is to build a report that shows the total spend for each account across each hour of the day.
+This means the timestamp column needs be be rounded down from millisecond to hour granularity. 
+
 Just like a SQL query, Flink can select the required fields and group by your keys.
```
Comment (twalthr) [inline docs/try-flink/table_api.md:194]: mention the term `Table API` and quickly explain what it is good for like `fluent DSL with JavaDocs and IDE support inspired by SQL`?

File: docs/try-flink/table_api.md [hunk 7]
```diff
@@ -31,482 +31,290 @@ The Table API in Flink is commonly used to ease the definition of data analytics
...
-import org.apache.flink.table.api._
-import org.apache.flink.table.api.bridge.scala.StreamTableEnvironment;
-import org.apache.flink.walkthrough.common.table._
-
-object SpendReport {
 
-    def main(args: Array[String]): Unit = {
-        val env = StreamExecutionEnvironment.getExecutionEnvironment
-        env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+Finally, go to [Grafana](http://localhost:3000/d/FOe0PbmGk/walkthrough?viewPanel=2&orgId=1&refresh=5s) to see the fully visualized result!
```
Comment (twalthr) [inline docs/try-flink/table_api.md:316]: I haven't tried it but is `FOe0PbmGk` the same for all users?

Comment (sjwiesman) [reply docs/try-flink/table_api.md:316]: Yes, I ran docker compose several times it is stable

File: docs/try-flink/table_api.md (hunk 2 above)
Comment (sjwiesman) [reply docs/try-flink/table_api.md:112]: true, that's a remnant from the old planner description. I'll rework it.

File: docs/try-flink/table_api.md (hunk 6 above)
Comment (sjwiesman) [reply docs/try-flink/table_api.md:194]: yep.

**Comment by sjwiesman** [comment]
@twalthr thank you for taking a look, I have pushed an update.

File: docs/try-flink/table_api.md [hunk 8]
```diff
@@ -183,20 +183,22 @@ TableEnvironment tEnv = TableEnvironment.create(settings);
...
 
 ## Attempt One
 
 Now with the skeleton of a Job set-up, you are ready to add some business logic.
 The goal is to build a report that shows the total spend for each account across each hour of the day.
 This means the timestamp column needs be be rounded down from millisecond to hour granularity. 
 
-Just like a SQL query, Flink can select the required fields and group by your keys.
+Flink supports developing relational applications in pure [SQL]({% link dev/table/sql/index.md %}) or using the [Table API]({% link dev/table/tableApi.md %}).
+The Table API is a fluent DSL inspired, that can be written in Python, Java, or Scala and supports strong IDE integration.
```
Comment (twalthr) [inline docs/try-flink/table_api.md:195]: `inspired by SQL`?

Comment (sjwiesman) [reply docs/try-flink/table_api.md:195]: Whoops, yes nice catch

**Pull request closed without merging**